src/ng/directive/ngOptions.js

```diff
@@ -618,10 +618,19 @@ var ngOptionsDirective = ['$compile', '$parse', function($compile, $parse) {
         var emptyOption_ = emptyOption && emptyOption[0];
         var unknownOption_ = unknownOption && unknownOption[0];
 
+        // If the compiled empty option was replaced by a comment because
+        // it had an "element" transclusion directive on it (such as ngIf)
+        // then compare against the pre-compiled empty element instead
+        if (emptyOption && emptyOption_.nodeType === NODE_TYPE_COMMENT) {
```

Comment:
If this block is removed, don't forget to also undo the change on [line 411][1].

[1]: https://github.com/angular/angular.js/pull/13033/files#diff-0a9e86a58b2663074f51454f7a18a0efR411